### PR TITLE
DEP: remove extra np.deprecate and add docs back

### PR DESCRIPTION
**Pull request by j-bowhay**
#### Reference issue

Follow up to https://github.com/scipy/scipy/pull/18861

### CODE REVIEW

File: scipy/signal/_bsplines.py [hunk 1]
```diff
@@ -346,9 +370,20 @@ def _cubic(x):
 """
 
 
-@np.deprecate(message=msg_quadratic)
```
Comment (j-bowhay) [inline scipy/signal/_bsplines.py:349]: This one slipped through

**Review by rgommers — COMMENTED** [review]
Thanks for this follow-up Jake. LGTM, but a little tweak is needed to make the refguide checker happy.

File: scipy/signal/_bsplines.py [hunk 2]
```diff
@@ -157,7 +156,20 @@ def thefunc(x):
...
+
+        `scipy.signal.bspline` is deprecated in SciPy 1.11 and will be
+        removed in SciPy 1.13.
+
+        The exact equivalent (for a float array `x`) is
+
+        >>> from scipy.interpolate import BSpline
+        >>> knots = np.arange(-(n+1)/2, (n+3)/2)
+        >>> out = BSpline.basis_element(knots)(x)
+        >>> out[(x < knots[0]) | (x > knots[-1])] = 0.0
```
Comment (j-bowhay) [inline scipy/signal/_bsplines.py:170]: Is there some directive or something I can use here so this code doesn't execute?

Comment (ev-br) [reply scipy/signal/_bsplines.py:170]: Why is any of this needed?

Comment (ev-br) [reply scipy/signal/_bsplines.py:170]: I mean, either the code snippet is incorrect and should be fixed (but why, what is the change that made it invalid?) or it is correct and the error is within the refguide-checker (which I'm happy to take a look at)

Comment (ev-br) [reply scipy/signal/_bsplines.py:170]: Ah, we now require explicit `>>> import numpy as np` in all examples, and this one is missing it. Also in `cubic` and `quadratic`.

Comment (j-bowhay) [reply scipy/signal/_bsplines.py:170]: Also `x` is not declared

Comment (ev-br) [reply scipy/signal/_bsplines.py:170]: Thanks Jake. In that case, indeed, this is not meant to be a complete example, so OK to hide from the doctester in whichever way works :-).

**Comment by j-bowhay** [comment]
The refguide check doesn't seem to respect literal blocks so I opted just to skip these functions in the refguide check. I think this is probably fine since they will be gone in 6 months.

Lint failure unrelated

**Pull request merged**